ui/plugins/failure_summary_panel.jsx

```diff
@@ -0,0 +1,220 @@
+    }
+
+    render() {
+        //If this method is bound in the constructor it gives me a warning about only setting state in a mounted component
+        // but if we move to allowing the arrow function in classes at some point, this problem should be solved.
+        this.clickShowMore = this.clickShowMore.bind(this);
+        return (
+            <li>
+                <div className="job-tabs-content">
+                    {this.props.filerInAddress || this.props.user.is_staff &&
+                    <a className="btn btn-xs btn-light-bordered"
+                       onClick={this.fileBugEvent}
+                       title="file a bug for this failure">
+                        <i className="fa fa-bug"></i>
+                    </a>}
+                    <span>{this.props.suggestion.search}</span>
+                </div>
+
+                {/* <!--Open recent bugs--> */}
+                {this.props.suggestion.valid_open_recent &&
+                <ul className="list-unstyled failure-summary-bugs">
+                    {this.props.suggestion.bugs.open_recent.map((bug, index) =>
+                        <BugListItem
+                                    key={index} bug={bug} selectedJob={this.props.selectedJob}
+                                    getBugUrl={this.props.getBugUrl} pinboardService={this.props.pinboardService}
+                                    escapeHTMLFilter={this.props.escapeHTMLFilter} suggestion={this.props.suggestion}
+                                    highlightCommonTermsFilter={this.props.highlightCommonTermsFilter}/>)}
+
+                </ul>}
+
+                {/* <!--All other bugs--> */}
+                {this.props.suggestion.valid_all_others && this.props.suggestion.valid_open_recent &&
+                <a target="_blank"
+                   onClick={this.clickShowMore}
+                   className="show-hide-more">Show / Hide more</a>}
+
+                {this.props.suggestion.valid_all_others && (this.state.suggestionShowMore
+                || !this.props.suggestion.valid_open_recent) &&
+                <ul className="list-unstyled failure-summary-bugs">
+                    {this.props.suggestion.bugs.all_others.map((bug, index) =>
+                        <BugListItem
+                                    key={index} bug={bug} selectedJob={this.props.selectedJob}
+                                    getBugUrl={this.props.getBugUrl} pinboardService={this.props.pinboardService}
+                                    escapeHTMLFilter={this.props.escapeHTMLFilter} suggestion={this.props.suggestion}
+                                    highlightCommonTermsFilter={this.props.highlightCommonTermsFilter}
+                                    bugClassName={bug.resolution !== "" ? "deleted" : ""} />)}
+                </ul>}
+
+                {this.props.suggestion.bugs.too_many_open_recent || (this.props.suggestion.bugs.too_many_all_others
+                && !this.props.suggestion.valid_open_recent) &&
```

Comment:
This also needs extra conditional isolation, such as:

    {(this.props.suggestion.bugs.too_many_open_recent || (this.props.suggestion.bugs.too_many_all_others
      && !this.props.suggestion.valid_open_recent)) &&

Reply:
Basically, any time we have an ``ng-if`` that has more than one conditional, especially an OR conditional, we should isolate it from the ``&&`` that is used for rendering.

Reply:
Ah, yes - I'll pay closer attention to operator precedence/short circuit logic. Also, I think I placed the parens incorrectly. Shouldn't it be: `{((this.props.suggestion.bugs.too_many_open_recent || this.props.suggestion.bugs.too_many_all_others)
  && !this.props.suggestion.valid_open_recent) &&` ?